### PR TITLE
[new-ui] Panels & themes

### PR DESCRIPTION
**Pull request by walaura**
Thinking out loud and writing docs about how to do themability [here](https://github.com/graphql/graphiql/blob/121fbfd75a5d1ffa948a73a819f286d8506bca7d/packages/graphiql/src/new-components/themes/README.md), comments welcome! 

I also added new components for flex layouts (super exciting) and pink titles (actually exciting)

And made it so storybook automatically generates jest snaps, this seems pretty sensible because they are basically the same thing.

### CODE REVIEW

File: babel.config.js [hunk 1]
```diff
@@ -3,30 +3,28 @@
 const envConfig = {
   ignoreBrowserslistConfig: true,
   modules: 'commonjs',
-  targets: { node: true }
```
Comment (walaura) [inline babel.config.js]: sorry, i accidentally prettified the babel config 😄

File: babel.config.js [hunk 2]
```diff
@@ -3,30 +3,28 @@
...
-      envConfig,
-    ],
+    [require.resolve('@babel/preset-env'), envConfig],
     require.resolve('@babel/preset-flow'),
     require.resolve('@babel/preset-react'),
   ],
   plugins: [
     require.resolve('@babel/plugin-proposal-class-properties'),
     require.resolve('@babel/plugin-syntax-dynamic-import'),
+    require.resolve('babel-plugin-macros'),
```
Comment (walaura) [inline babel.config.js]: we really only want this in the environments where we run jest, not prod. it doesn't hurt to have it around but it's messy to not remove it. how do we do that?

Comment (acao) [reply babel.config.js]: good question, lets add it to the babel config itself like this:
```js
env: {
    test: {
      plugins: [require.resolve('babel-plugin-macros')]
    }
  },
plugins: [
  require.resolve('@babel/plugin-proposal-class-properties'),
  ...
]
```
jest should automatically set `process.env.BABEL_ENV=test`

File: babel.config.js (hunk 1 above)
Comment (acao) [reply babel.config.js]: haha no problem! thats a good thing

**Comment by acao** [comment]
@walaura thanks a bunch, this is looking so splendid! giving me all kinds of ideas for the frontbackend side of things, seeing things come to fruition on this side

**Comment by walaura** [comment]
yas!!! so excited to hear

File: packages/graphiql/src/new-components/themes/default/Layout.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+          <Card>{explorer.input}</Card>
+          <Card>{explorer.response}</Card>
+          <Card innerSx={{ gridArea: 'console' }}>{explorer.console}</Card>
+        </div>
+      )}
+    </main>
+  );
+};
+
+Layout.propTypes = LAYOUT_PROP_TYPES;
```
Comment (acao) [inline packages/graphiql/src/new-components/themes/default/Layout.js]: this makes perfect sense for now, but as you may know we are converting to typescript within a month ish by very popular demand. once we get the some things implemented

File: packages/graphiql/src/new-components/themes/default/index.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
+import Layout from './Layout';
+
+const palette = {
+  neutral: {
+    20: '#939393',
+    90: `rgba(0, 0, 0, 0.1)`,
+    100: '#fff',
+  },
+  pink: {
```
Comment (acao) [inline packages/graphiql/src/new-components/themes/default/index.js]: i think @orta wanted to call this fuscia? oooh but how will we officially pronounce fuscia? sounds like a standards body decision

File: packages/graphiql/src/new-components/themes/default/Layout.js (hunk 3 above)
Comment (walaura) [reply packages/graphiql/src/new-components/themes/default/Layout.js]: omg honestly can't wait!! i feel a bit like freefalling in here not having typechecking. Can I use TS on this part already?

File: packages/graphiql/src/new-components/themes/default/index.js (hunk 4 above)
Comment (walaura) [reply packages/graphiql/src/new-components/themes/default/index.js]: let's

**Comment by acao** [comment]
looks good to merge! whatchu think @walaura

**Pull request merged**